analyzers/tests/SonarAnalyzer.TestFramework/Verification/IssueValidation/IssueLocationCollector.cs

```diff
@@ -50,19 +50,19 @@ internal static class IssueLocationCollector
         private static readonly Regex RxInvalidType = CreateRegex(CommentPattern + ".*" + IssueTypePattern);
         private static readonly Regex RxInvalidPreciseLocation = CreateRegex(@"^\s*" + CommentPattern + ".*" + PrecisePositionPattern);
 
-        public static IList<IssueLocation> GetExpectedIssueLocations(IEnumerable<TextLine> lines)
+        public static IList<IssueLocation> GetExpectedIssueLocations(string filePath, IEnumerable<TextLine> lines)
         {
             var preciseLocations = new List<IssueLocation>();
             var locations = new List<IssueLocation>();
 
             foreach (var line in lines)
             {
-                var newPreciseLocations = GetPreciseIssueLocations(line).ToList();
+                var newPreciseLocations = GetPreciseIssueLocations(filePath, line).ToList();
                 if (newPreciseLocations.Any())
                 {
                     preciseLocations.AddRange(newPreciseLocations);
                 }
-                else if (GetIssueLocations(line).ToList() is var newLocations && newLocations.Any())
+                else if (GetIssueLocations(filePath, line).ToList() is var newLocations && newLocations.Any())
```

Comment:
I don't want to touch IssueLocationCollector this sprint too much. That's a whole new level of rabbit hole. 

The ToArray cannot go away, because the inner parts are coming from a `Select` and it's iterated twice (.Any(), AddRange)